### DISCUSSION TITLE
Question to running this lib on ESP32 S2

### DISCUSSION THREAD

**Discussion by Linzm99** [Dlloydev/ESP32-ESP32S2-AnalogWrite#11] [Q&A] [answered]
Hi, first of all: This library is a really good adition for integrating better ESP32 Support to the Arduino IDE!
But I have a Problem. I am trying to run this lib on an ESP 32 S2 and I used multiple different functions and even when uploading the example code it doesnt work. Did I forget to configure something?

**Comment by Dlloydev** [comment, marked as answer]
Thanks,

I would first of all check you have the correct library. There's another library with similar name (`ESP32_AnalogWrite`) by ERROPiX that's easy to download and use by mistake.

For this library, you'll need to use the latest boards manager installation of the Arduino core for the ESP32. It should show version 2.0.0-rc1

![image](https://user-images.githubusercontent.com/63488701/130356515-391c266c-24ac-42c4-bf8a-3f3f147a73c7.png)
If you use their latest code from GitHub it will have a fix that would cause incorrect operation of some functions. See #9

**Reply by Linzm99** [reply]
Hi, thanks for the fast reply. I can only select 1.0.5 as version. I can remember that I installed the latest version for esp32 s2 support not over the board manager but by changing files by hand. Any suggerstions or good tutorial how to get that latest verion?

Edit: Never mind I got it

**Reply by Linzm99** [reply]
I installed version 2.0.0-rc1 but my ESP32-S2 is still not generating any PWM on PIN 4 when using: analogWrite(4, 341, 100, 10, 341);

Edit: I literally changed nothing but uploaded the code several Times and now it works ...

**Reply by Dlloydev** [reply]
Glad its now working!
